R/download_data_wrds_compustat.R

```diff
@@ -85,9 +89,40 @@ download_data_wrds_compustat <- function(type, start_date, end_date, additional_
         inv = at / at_lag - 1,
         inv = if_else(at_lag <= 0, NA, inv)
       )
+  }
+
+  if (grepl("compustat_quarterly", type, fixed = TRUE)) {
+    fundq_db <- tbl(con, in_schema("comp", "fundq"))
+
+    compustat <- fundq_db |>
+      filter(
+        indfmt == "INDL" &
+          datafmt == "STD" &
+          consol == "C" &
+          between(datadate, start_date, end_date)
+      ) |>
+      select(
+        gvkey, datadate, rdq, fqtr, fyearq,
+        all_of(additional_columns)
+      ) |>
+      collect()
 
     disconnection_connection(con)
 
-    processed_data
+    compustat <- compustat |>
+      drop_na(gvkey, datadate, fyearq, fqtr)|>
+      mutate(date = ceiling_date(datadate, "quarter") %m-% months(1)) |>
+      group_by(gvkey, fyearq, fqtr) |>
+      filter(datadate == max(datadate)) |>
+      slice_head(n = 1) |>
+      ungroup() |>
+      filter(if_else(is.na(rdq), TRUE, date < rdq))
```

Comment:
Move this filter up to maybe reduce cases that are dropped in the grouped filter.